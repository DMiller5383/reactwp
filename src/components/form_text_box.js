import React, {Component} from 'react';
import _ from 'lodash';


export const formTextBox  = (field) {
  return(
    <div className="input-row">
      <input {...field.input} type="text"/>
      {field.meta.touched && field.meta.error && <span className="error">{field.meta.error}</span>}
    </div>
  )
}
