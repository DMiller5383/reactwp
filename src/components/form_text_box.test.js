import React, {Component} from 'react';
import {FormTextBox} from './form_text_box';
import {shallow, render, wrapper} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('form_text_box.js', ()=>{
  let field =  {
    input: {
      name: 'title'
    },
    meta: {
      touched: false,
      error: ''
    },
  }

  let fieldWithErrors = {
    input: {
      name: 'title'
    },
    meta: {
      touched: true,
      error: 'There was an error.'
    },
  }
  let textBox;
  let textBoxWithErrors;
  beforeEach(()=>{
    textBox = shallow(<FormTextBox input={field.input} meta={field.meta}  />);
    textBoxWithErrors = shallow(<FormTextBox input={fieldWithErrors.input} meta={fieldWithErrors.meta} />);
  })
  it('renders correctly', ()=>{
    expect(toJson(textBox)).toMatchSnapshot();
  });

  it('renders a textbox', ()=>{
    let textBoxCount = textBox.find('input').length;
    expect(textBoxCount).toBe(1);
  });

  it('will display errors if there are any', ()=>{
    let error = textBoxWithErrors.find('.error');
    expect(error.text()).toBe('There was an error.');
  });

  it('will not display errors if touched is false', ()=>{

  });
});
