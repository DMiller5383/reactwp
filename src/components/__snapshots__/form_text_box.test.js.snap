// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form_text_box.js renders correctly 1`] = `
<div
  className="input-row"
>
  <input
    name="title"
    type="text"
  />
</div>
`;
